ARROW-13401 [R]: Update python.Rmd vignette (#11914)
apache/arrow: r/vignettes/python.Rmd

  %\VignetteEncoding{UTF-8}
---

The arrow package provides [reticulate](https://rstudio.github.io/reticulate/) methods for passing data between
R and Python in the same process. This document provides a brief overview.

> Why you might want to use `pyarrow`?
> * To use some Python functionality that is not yet implemented in R, for example, the `concat_arrays` function.
> * To transfer Python objects into R, for example, a Pandas dataframe into an R Arrow Array. 

Reviewer: ```suggestion
Why you might want to use `pyarrow`?

* To use some Python functionality that is not yet implemented in R, for example, the `concat_arrays` function.
* To transfer Python objects into R, for example, a Pandas dataframe into an R Arrow Array. 
```
Needs a space to render properly

Author: Ok, I see I need to build the docs next time, sorry for giving you additional work! I wanted to try how it goes ...

> The `arrow` package provides `reticulate` methods for passing data between

Reviewer: ```suggestion
The arrow package provides [reticulate](https://rstudio.github.io/reticulate/) methods for passing data between
```

pa <- import("pyarrow")
```

> The pyarrow package includes support for sharing Arrow `Array` and `RecordBatch`

Reviewer: ```suggestion
The arrow R package include support for sharing Arrow `Array` and `RecordBatch`
```

Author: Heh, got that one wrong!

pa <- import("pyarrow")
```

> The `pyarrow` package includes support for sharing Arrow `Array` and `RecordBatch`

Reviewer: ```suggestion
The pyarrow package includes support for sharing Arrow `Array` and `RecordBatch`
```

objects in-process between R and Python. For example, let's create an `Array`

> in `pyarrow`.

Reviewer: ```suggestion
in pyarrow.
```

## ]
```

> `a` is now an `Array` object in our R session, even though you created it in Python.

Reviewer: ```suggestion
`a` is now an `Array` object in your R session, even though you created it in Python.
```

## ]
```

> You can send data both ways. One reason you might want to use `pyarrow` in R is

Reviewer: ```suggestion
You can send data both ways. One reason you might want to use pyarrow in R is
```

to take advantage of functionality that is better supported in Python than in R.

> For example, `pyarrow` has a `concat_arrays` function, but as of 0.17, this

Reviewer: ```suggestion
For example, pyarrow has a `concat_arrays()` function, but as of 0.17, this
```

> function is not implemented in the `arrow` R package. You can use `reticulate`

Reviewer: ```suggestion
function is not implemented in the arrow R package. You can use reticulate
```

## ]
```

> Now you have a single `Array` in R.

Reviewer: ```suggestion
Now you have a single Array in R.
```

## How this works

"Send", however, isn't the correct word. Internally, we're passing pointers to
the data between the R and Python interpreters running together in the same
process, without copying anything. Nothing is being sent: we're sharing and
accessing the same internal Arrow memory buffers.

## Arrow object types

> For more information about Arrow object types see the Internals section of

Reviewer: ```suggestion
For more information about Arrow object types see the "Internals" section of
```

> the Arrow vignette by calling:

Reviewer: ```suggestion
the "arrow" vignette:
```
